### PR TITLE
improve endpoint support

### PR DESCRIPTION
**Pull request by drbh**
small PR to add a new interface endpoint behind a feature

### CODE REVIEW

File: router/src/main.rs [hunk 1]
```diff
@@ -328,6 +328,15 @@ async fn main() -> Result<(), RouterError> {
     tracing::info!("Setting max batch total tokens to {max_supported_batch_total_tokens}");
     tracing::info!("Connected");
 
+    // Determine the server port based on the feature and environment variable.
+    let port = if cfg!(feature = "google") {
```
Comment (Narsil) [inline router/src/main.rs:332]: Let's make that compile time.

```rust
#[cfg(feature = "google")]
let port = xxx

#[cfg(not(feature = "google"))]
let port = xxxx
```

File: router/src/server.rs [hunk 2]
```diff
@@ -969,10 +1058,24 @@ pub async fn run(
...
 
     // Combine routes and layers
-    let app = Router::new()
+    let mut app = Router::new()
         .merge(swagger_ui)
         .merge(base_routes)
-        .merge(aws_sagemaker_route)
+        .merge(aws_sagemaker_route);
+
+    if cfg!(feature = "google") {
```
Comment (Narsil) [inline router/src/server.rs:1066]: Same, compile time is preferred I think.

File: router/src/main.rs (hunk 1 above)
Comment (Narsil) [reply router/src/main.rs:332]: Nvm, I checked the macros, it's also compile time.

Comment (drbh) [reply router/src/main.rs:332]: nice, however if helpful I can update the syntax

**Comment by Narsil** [comment]
Have you checked if the openapi is correctly created ? I think you forgot to add the structs to the utopia thing to generate them  properly, no ?

**Review by Narsil — APPROVED** [review]

**Comment by OlivierDehaene** [comment]
This PR does not compile.

File: router/src/server.rs [hunk 3]
```diff
@@ -942,8 +1036,30 @@ pub async fn run(
...
+        #[cfg(feature = "google")]
+        {
+            // limiting mutability to the smallest scope necessary
+            let mut doc = doc;
+            doc.merge(VertextApiDoc::openapi());
+            doc
+        }
+        #[cfg(not(feature = "google"))]
+        ApiDoc::openapi()
+    };
```
Comment (OlivierDehaene) [inline router/src/server.rs:1059]: This should be:

```rust
// Define VertextApiDoc conditionally only if the "google" feature is enabled
    let doc = {
        // avoid `mut` if possible
        #[cfg(feature = "google")]
        {
            use crate::VertexInstance;

            #[derive(OpenApi)]
            #[openapi(
            paths(vertex_compatibility),
            components(schemas(VertexInstance, VertexRequest, VertexResponse))
            )]
            struct VertextApiDoc;

            // limiting mutability to the smallest scope necessary
            let mut doc = ApiDoc::openapi();
            doc.merge(VertextApiDoc::openapi());
            doc
        }
        #[cfg(not(feature = "google"))]
        ApiDoc::openapi()
    };
```

**Pull request merged**